app/main/views/outcome.py

```diff
@@ -48,36 +46,32 @@ def award_or_cancel_brief(framework_slug, lot_slug, brief_id):
         )
     }])
 
-    if brief['status'] in ["awarded", "cancelled", "unsuccessful"]:
-        already_awarded = True
-    else:
-        already_awarded = False
-
-        if request.method == "POST":
-            form = AwardOrCancelBriefForm(brief, request.form)
-            if not form.validate_on_submit():
-                errors = get_errors_from_wtform(form)
-            else:
-                answer = form.data.get('award_or_cancel_decision')
-                if answer == 'back':
-                    flash(BRIEF_UPDATED_MESSAGE.format(brief=brief))
-                    return redirect(url_for('.buyer_dos_requirements'))
-                elif answer == 'yes':
-                    return redirect(
-                        url_for('.award_brief', framework_slug=framework_slug, lot_slug=lot_slug, brief_id=brief_id)
-                    )
-                elif answer == 'no':
-                    return redirect(url_for(
-                        '.cancel_award_brief', framework_slug=framework_slug, lot_slug=lot_slug, brief_id=brief_id)
-                    )
-                else:
-                    # We should never get here as the form validates the answers against the available choices.
-                    abort(500, "Unexpected answer to award or cancel brief")
+    form = AwardOrCancelBriefForm(brief)
+    already_awarded = brief['status'] in ["awarded", "cancelled", "unsuccessful"]
+
+    if already_awarded is False and form.validate_on_submit():
+        answer = form.data.get('award_or_cancel_decision')
+        if answer == 'back':
+            flash(BRIEF_UPDATED_MESSAGE.format(brief=brief))
+            return redirect(url_for('.buyer_dos_requirements'))
+        elif answer == 'yes':
+            return redirect(
+                url_for('.award_brief', framework_slug=framework_slug, lot_slug=lot_slug, brief_id=brief_id)
+            )
+        elif answer == 'no':
+            return redirect(url_for(
+                '.cancel_award_brief', framework_slug=framework_slug, lot_slug=lot_slug, brief_id=brief_id)
+            )
+        else:
+            # We should never get here as the form validates the answers against the available choices.
+            abort(500, "Unexpected answer to award or cancel brief")
+
+    errors = get_errors_from_wtform(form)
 
     return render_template(
         "buyers/award_or_cancel_brief.html",
         brief=brief,
-        form=form or AwardOrCancelBriefForm(brief),
+        form=form,
```

Comment:
Lovely. Could we do the same for `award_brief` and `cancel_brief`? 😸

Reply:
😢

Reply:
Done 👍